C/28.c

```diff
@@ -0,0 +1,23 @@
+#include <stdio.h>
+int main() {
+    int n, i, k=0;
+    printf("Enter a positive integer: ");
+    scanf("%d", &n);
+ for (i=2;i<=n/2;i++) {
+        if (n%i==0) {
+            k=1;
+            break;
+        }
+    }
+    if (n==1) {
```

Comment:
@ssj1901 Move this `n == 1` logic before the `for loop` as it doesn't make any sense to go through the loop if `n==1` and then update the conditional statement for the variable k.